### PR TITLE
 [ci skip] Append some description about advance constraints

### PR DESCRIPTION
**Pull request by lanzhiheng**
### Summary

Simple changes of the document for better describing the usage of routes's constraints.

### CODE REVIEW

File: guides/source/routing.md [hunk 1]
```diff
@@ -751,6 +751,35 @@ end
 
 Both the `matches?` method and the lambda gets the `request` object as an argument.
 
+#### advance in a block form
```
Comment (eileencodes) [inline guides/source/routing.md:754]:
```suggestion
#### Constraints in block form
```

File: guides/source/routing.md [hunk 2]
```diff
@@ -751,6 +751,35 @@ end
...
+
+Of course, you can specify this kind of constraints in a block form. Especially, when you need to apply same rule to several routes. Just like
```
Comment (eileencodes) [inline guides/source/routing.md:756]:
```suggestion
You can specify constraints in a block form. This is useful for when you need to apply the same rule to several routes. For example
```

File: guides/source/routing.md [hunk 3]
```diff
@@ -751,6 +751,35 @@ end
...
+
+Rails.application.routes.draw do
+  constraints(RestrictedListConstraint.new) do
+    get '*path', to: 'restricted_list#index',
+    get '*other-path', to: 'other_restricted_list#index',
+  end
+end
+```
+
+You also can apply `lambda` version.
```
Comment (eileencodes) [inline guides/source/routing.md:771]:
```suggestion
You also use a `lambda`:
```

**Review by eileencodes — CHANGES_REQUESTED** [review]
I've added some suggestions to clarify the language.

**Comment by lanzhiheng** [comment]
@eileencodes Thanks for your advice, I have updated all of them.

**Review by eileencodes — APPROVED** [review]
Thanks for making the changes, and apologies for the delay in re-reviewing.

**Pull request merged**